systemvm/debian/opt/cloud/bin/cs_dhcp.py

```diff
@@ -28,7 +28,7 @@ def merge(dbag, data):
     else:
         remove_keys = set()
         for key, entry in dbag.iteritems():
-            if key != 'id' and entry['mac_address'] == data['mac_address'] and data['remove']:
+            if key != 'id' and entry['mac_address'] == data['mac_address']:
```

Comment:
@weizhouapache could this cause any regression it it always removes keys? See old commits d62d5c6cd23daeb922e5c6f64399cc30596c15a3, 8fb388e9312b917a8f36c7d7e3f45985a95ce773, c9ce3e2344ab863dae09e63bf783331097a660cb

Could it break for:
- stop and destroy a VM - does its ip/dhcp/dns rules get removed or not?
- add/remove a new nic to VM (multi-homed vms in VPC)
- update/replace new IPs to VM, say add secondary IPs to a VM
- change in nic/ip order?

Reply:
@rhtyd thanks for remarks.
I have tested some user actions (except secondary ips), all looks good until now.
I will test more actions manually.